src/Controllers/MovieController.cs

```diff
@@ -29,71 +30,76 @@ private IUnitOfWork UnitOfWork
         }
 
         [HttpGet(""), Authorize]
-        public IActionResult Index(string searchString, string movieGenre)
+        public IActionResult Index(string searchString, string movieGenre, string sortOrder)
         {
             try
             {
+                ViewData["TitleSort"] = String.IsNullOrEmpty(sortOrder) ? "title_desc" : "";
+                ViewData["DateSort"] = sortOrder == "date" ? "date_desc" : "date";  
+                ViewData["GenreSort"] = sortOrder == "genre" ? "genre_desc" : "genre";
+                ViewData["PriceSort"] = sortOrder == "price" ? "price_desc" : "price";
+                ViewData["RatingSort"] = sortOrder == "rating" ? "rating_desc" : "rating";
                 var movies = UnitOfWork.MovieRepository.GetAll();
-                var genreQuery = movies.OrderBy(m => m.Genre).Select(m => m.Genre).Distinct().ToList();
+                var movieGenreVM = new MovieGenreViewModel();
                 if (movies == null)
+                {
                     throw new NullReferenceException();
+                }
+                var genreQuery = movies.OrderBy(m => m.Genre).Select(m => m.Genre).Distinct().ToList();
+
                 if (!String.IsNullOrEmpty(searchString))
                 {
-                    var moviesVM = movies.Select(m => new MovieViewModel
+                    movies = movies.Where(m => m.Title.Contains(searchString));
```

Comment:
There's no need to use brackets when if statements have only one line
```csharp
    if (!String.IsNullOrEmpty(searchString))
        movies = movies.Where(m => m.Title.Contains(searchString));
```

Reply:
I thought it was the best way to clarify which line belongs to each "if"

Reply:
It's a valid point, but indentation should be enough to recognize those statements